community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/Effects.scala

```diff
@@ -108,39 +125,36 @@ object Effects {
 }
 
 trait Effect {
-  def reads: Boolean
-
-  def writes: Boolean
 
   override def toString = this.getClass.getSimpleName
 }
 
-protected trait ReadEffect extends Effect {
-  override def reads = true
+protected trait ReadEffect extends Effect
+
+protected trait WriteEffect extends Effect
 
-  override def writes = false
+case class LeafEffect(effect: Effect) extends Effect {
 
-  def toWriteEffect: WriteEffect
+  override def toString = s"${super.toString} (${effect.toString})"
```

Comment:
Why are we overriding toString, it doesn't give us anything that is not already in the default case class toString?